src/wsteth-exchange-rate-adapter/WstEthEthExchangeRateChainlinkAdapter.sol

```diff
@@ -15,17 +15,17 @@ contract WstEthEthExchangeRateChainlinkAdapter is MinimalAggregatorV3Interface {
     uint8 public constant decimals = 18;
     string public constant description = "wstETH/ETH exchange rate";
 
-    IWstEth public immutable WST_ETH;
+    IStEth public immutable ST_ETH;
 
-    constructor(address wstEth) {
-        require(wstEth != address(0), ErrorsLib.ZERO_ADDRESS);
+    constructor(address stEth) {
+        require(stEth != address(0), ErrorsLib.ZERO_ADDRESS);
 
-        WST_ETH = IWstEth(wstEth);
+        ST_ETH = IStEth(stEth);
     }
 
-    /// @dev Silently overflows if `stEthPerToken` is greater than `type(int256).max`.
+    /// @dev Silently overflows if `getPooledEthByShares`'s return value is greater than `type(int256).max`.
     function latestRoundData() external view returns (uint80, int256, uint256, uint256, uint80) {
-        // It is assumed that `stEthPerToken` returns a price with 18 decimals precision.
-        return (0, int256(WST_ETH.stEthPerToken()), 0, 0, 0);
+        // It is assumed that `getPooledEthByShares` returns a price with 18 decimals precision.
+        return (0, int256(ST_ETH.getPooledEthByShares(1 ether)), 0, 0, 0);
```

Comment:
Again, I find it weird that the implementation relies on the fact that the precision of the feed is 18. If I fork the oracle and change the "decimals", it is not working anymore.

This is safer:
```suggestion
        return (0, int256(ST_ETH.getPooledEthByShares(1 ether) * 10 ** (decimals - 18)), 0, 0, 0);
```

nb: now that we hardcoded the stETH address, it's ok to hardcode the wstETH decimals

Reply:
>  If I fork the oracle and change the "decimals", it is not working anymore

yeah but we can say that for a lot of things no? 😅

Since the code is only valid for mainnet I don't think this is useful at all

Reply:
>yeah but we can say that for a lot of things no?

not but i mean a "parameter" of the contract

like if you change the virtual shares in blue, it's still supposed to work

I see what you mean, it doesn't have to be seen as a parameter but as a getter. 

But I'm not sure that's great

Reply:
I'm still fine to keep it as is, because `decimals` is a constant here

Reply:
> it doesn't have to be seen as a parameter but as a getter

👌

Reply:
hmm ok, it seems that there is almost consensus on keeping the code as is. Could we at least add a comment before decimal's definition, so explicitly say that the implem always return something with 18 decimals (whatever the value that is written there)?